nixos/modules/services/networking/networkmanager.nix

```diff
@@ -446,6 +463,85 @@ in
             will be substituted into the static configuration file using [envsubst](https://github.com/a8m/envsubst).
           '';
         };
+        secrets = {
```

Comment:
I think I would prefer this to go to a separate module file since it is not part of NetworkManager. The module is already too big for my liking.